src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/FilingsService.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.limitedpartnershipsapi.exception.ServiceException;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.generalpartner.dto.GeneralPartnerDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.limitedpartner.dto.LimitedPartnerDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.partnership.dto.LimitedPartnershipDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.utils.ApiLogger;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.FILING_KIND_LIMITED_PARTNERSHIP;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.GENERAL_PARTNER_FIELD;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.LIMITED_PARTNERSHIP_FIELD;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.LIMITED_PARTNER_FIELD;
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.TRANSACTION_KEY;
+
+@Service
+public class FilingsService {
+
+    private final LimitedPartnershipService  limitedPartnershipService;
+    private final GeneralPartnerService generalPartnerService;
+    private final LimitedPartnerService limitedPartnerService;
+
+    @Autowired
+    public FilingsService(LimitedPartnershipService limitedPartnershipService,
+                          GeneralPartnerService generalPartnerService,
+                          LimitedPartnerService limitedPartnerService){
+
+        this.limitedPartnershipService = limitedPartnershipService;
+        this.generalPartnerService = generalPartnerService;
+        this.limitedPartnerService = limitedPartnerService;
+    }
+
+    public FilingApi generateLimitedPartnerFilings(Transaction transaction) throws ServiceException {
```

Comment:
It's only one CHIPS filing we are generating here so suggest removing the 's' off the end of this method name.